libcodechecker/libhandlers/analyze.py

```diff
@@ -569,9 +520,9 @@ def main(args):
                       "exist!")
             continue
 
-        parseLogOptions = ParseLogOptions(args)
-        actions += log_parser.parse_log(log_file, parseLogOptions,
-                                        skip_handler)
+        actions += log_parser.parse_log(
```

Comment:
You no longer pass the skip_handler to the parser. What is the reason? We do not want to process the build actions for skipped files. Skippink this work was a huge performance gain for chromium, see https://github.com/Ericsson/codechecker/pull/1790 for details.

Reply:
Sorry, this is my fault, I'll fix it. However, it is strange, how could the tests pass without skip handler.

Reply:
I got it. Skip handler is passed to the analyzer manager a few lines below so it is not necessary to skip the actions at this point.

Reply:
The reason why we want to do the skipping earlier is because we do not even want to create the build action if we will not use it anyways. It does take significant amount of time to create the build action, so this is an optimization rather than a functional feature. See the pull request I linked for details. 

Quote from the patch author: "It was always surprising me why CodeChecker first calls clang-tidy for Chromium project only on ~15th minute after start (my PC: core-i7 8700K, 16GB RAM, Samsung 970 SSD M.2). And now I know the answer - a lot of useless work while parsing options :-) Now the same takes ~1 minute and for "-j8" about 5Gb instead of 9Gb for "-j4" before (I skip thousands of autogenerated files, tests etc)."

Reply:
Just to summarize, I think we do need the skip handler here to skip source files early and avoid expensive command line parsing, but we do need skip handler later on as well, to be able to skip storing reports from headers that are included from non-skipped cpp files.

Reply:
These are not good news. The goal would be to provide this log parser module as a 3rd party vendor project so it can be used as a standalone tool too. So in the longer terms all libcodechecker dependencies should be eliminated. I'll figure out something...

Reply:
We could make it an optional dependency. If we use the tool from CodeChecker using the API we could have skip_handler as an optional parameter, but it would not be exposed when used from command line.